primitives/api/src/lib.rs

```diff
@@ -734,5 +734,13 @@ decl_runtime_apis! {
 	pub trait Metadata {
 		/// Returns the metadata of a runtime.
 		fn metadata() -> OpaqueMetadata;
+
+		/// Returns the metadata at a given version.
+		fn metadata_at_version(version: u32) -> Option<OpaqueMetadata>;
+
+		/// Returns the supported versions of metadata.
+		///
+		/// This can be used to call `metadata_at_version`.
```

Comment:
```suggestion
		/// Returns the available metadata versions.
```

Nit; Just because "This can be used to call.." felt a bit unclear :)

Reply:
I would keep "supported". Feels more clear on what this is about.